eng/build.ps1

```diff
@@ -670,10 +670,12 @@ function Setup-IntegrationTestRun() {
 }
 
 function Prepare-TempDir() {
-  $env:TEMP=$TempDir
-  $env:TMP=$TempDir
+  subst t: $TempDir
```

Comment:
This isn't checking the return code of `subst` and it can fail if `T:` is already taken.

Did we consider deleting the `subst` at the end here as we do in other scripts like determinism?

Reply:
> This isn't checking the return code of subst and it can fail if T: is already taken.

This code only runs in CI, so the drive shouldn't be taken. I can check the return code and fail the build in this case (how?).

> Did we consider deleting the subst at the end here as we do in other scripts like determinism?

Happy to do that, but not sure where?

Reply:
➡️ OK, I believe both issues are now corrected

Reply:
You should be able to use the pattern which will do the proper return code checking

```ps1
  Exec-Command "subst" "T: $TempDir"
```

> Happy to do that, but not sure where?

I think we need to have a `Unprepare-TempDir` we call in the `finally` under the same conditions we call `Prepare-TempDir`